src/me/corriekay/pokegoutil/windows/PokemonTab.java

```diff
@@ -381,6 +387,58 @@ private void powerUpSelected() {
         }
     }
 
+
+    //feature added by Ben Kauffman
+    private void toggleFavoriteSelected() {
+        ArrayList<Pokemon> selection = getSelectedPokemon();
+        if(selection.size() > 0) {
+            if(confirmOperation("Toggle Favorite", selection)) {
+                MutableInt err = new MutableInt(), success = new MutableInt(), total = new MutableInt(1);
+                selection.forEach(poke -> {
+                    try {
+                        System.out.println("Toggling favorite " + total.getValue() + " of " + selection.size());
+                        total.increment();
+                        SetFavoritePokemonResponseOuterClass.SetFavoritePokemonResponse.Result result = poke.setFavoritePokemon(!poke.isFavorite());
+                        System.out.println("Attempting to set favorite for " + PokeHandler.getLocalPokeName(poke) + " to " + !poke.isFavorite() + "...");
+                        go.getPlayerProfile().updateProfile();
+
+                        if(result == SetFavoritePokemonResponseOuterClass.SetFavoritePokemonResponse.Result.SUCCESS) {
+                            System.out.println(
+                                    "Favorite for " + PokeHandler.getLocalPokeName(poke) + " set to " + !poke.isFavorite() + ", Result: Success!");
+                            success.increment();
+
+                        } else {
+                            err.increment();
+                            System.out.println(
+                                    "Error toggling favorite for " + PokeHandler.getLocalPokeName(poke) + " , result: " + result.toString());
+                        }
+
+                    } catch (Exception e) {
+
+                        // if the error was "Contents of buffer are null"
+                        // it was likely successful
+                        // https://github.com/Grover-c13/PokeGOAPI-Java/issues?utf8=%E2%9C%93&q=Contents%20of%20buffer%20are%20null%20
+                        err.increment();
+                        System.out.println("Error toggling Pokémon(s) favorite! " + e.getMessage());
```

Comment:
Can you add the Name of the pokémon to that Error message, like in all other bulk operations in one of the latest bug fixes?
Just include that `PokeHandler.getLocalPokeName(poke)`.